### PR TITLE
Implement Force/Release for FLI

### PR DESCRIPTION
**Pull request by ktbarrett**

### CODE REVIEW

File: cocotb/share/lib/fli/FliObjHdl.cpp [hunk 1]
```diff
@@ -249,14 +249,41 @@ int FliEnumObjHdl::set_signal_value(const int32_t value,
...
     if (m_is_var) {
-        mti_SetVarValue(get_handle<mtiVariableIdT>(),
-                        static_cast<mtiLongT>(value));
+        switch (action) {
+            case GPI_DEPOSIT:
+                mti_SetVarValue(get_handle<mtiVariableIdT>(),
+                                static_cast<mtiLongT>(value));
+                return 0;
+            case GPI_FORCE:
+                LOG_ERROR("Cannot force a VHDL variable");
```
Comment (cmarqu) [inline cocotb/share/lib/fli/FliObjHdl.cpp:258]: Maybe reword these messages to
```suggestion
                LOG_ERROR("Forcing VHDL variables is not supported by FLI");
```
to prevent users reading this as a message about a specific variable (might happen if they are not native speakers).

File: tests/test_cases/test_force_release/test_force_release.py [hunk 2]
```diff
@@ -8,8 +8,6 @@
 # Questa FLI GPI implementation does not support Force/Release (gh-1855)
```
Comment (cmarqu) [inline tests/test_cases/test_force_release/test_force_release.py:8]:
```suggestion
```

File: cocotb/share/lib/fli/FliObjHdl.cpp (hunk 1 above)
Comment (ktbarrett) [reply cocotb/share/lib/fli/FliObjHdl.cpp:258]: Perhaps instead:
```suggestion
                LOG_ERROR("Cannot force VHDL variables");
```

VHDL variables can only be set, they do not support inertial writes or force/release in the language. IDK how the VHPI handles these cases, perhaps instead we should treat force as a regular variable set so it's consistent?

File: cocotb/share/lib/fli/FliObjHdl.cpp [hunk 3]
```diff
@@ -357,30 +406,93 @@ int FliLogicObjHdl::set_signal_value(const int32_t value,
...
         } else {
-            mti_SetSignalValue(get_handle<mtiSignalIdT>(),
-                               (mtiLongT)m_mti_buff);
+            switch (action) {
+                case GPI_DEPOSIT:
+                    mti_SetSignalValue(get_handle<mtiSignalIdT>(),
+                                       (mtiLongT)m_mti_buff);
+                    return 0;
+                case GPI_FORCE: {
+                    std::string value_str = "2#";
```
Comment (benbr8) [inline cocotb/share/lib/fli/FliObjHdl.cpp:431]: You could also use `10#<val>` and `-10#<val>`. Questa won't make a difference, since it will create an int from it anyway. Not sure which is better, just a note :).

**Review by benbr8 — APPROVED** [review]

File: cocotb/share/lib/fli/FliObjHdl.cpp (hunk 1 above)
Comment (marlonjames) [reply cocotb/share/lib/fli/FliObjHdl.cpp:258]: VHPI allows forcing variables. Section 22.5.2 of the LRM has details.

Comment (ktbarrett) [reply cocotb/share/lib/fli/FliObjHdl.cpp:258]: So there is no language-level support for forcing variables, but you can do it with the VHPI? Okay. I guess the first suggestion is more correct.

File: cocotb/share/lib/fli/FliObjHdl.cpp (hunk 3 above)
Comment (ktbarrett) [reply cocotb/share/lib/fli/FliObjHdl.cpp:431]: Does the force command accepts negative numbers for setting std_logic_vectors?

File: cocotb/share/lib/fli/FliObjHdl.cpp [hunk 4]
```diff
@@ -249,14 +249,42 @@ int FliEnumObjHdl::set_signal_value(const int32_t value,
...
     if (m_is_var) {
-        mti_SetVarValue(get_handle<mtiVariableIdT>(),
-                        static_cast<mtiLongT>(value));
+        switch (action) {
+            case GPI_DEPOSIT:
+                mti_SetVarValue(get_handle<mtiVariableIdT>(),
+                                static_cast<mtiLongT>(value));
+                return 0;
+            case GPI_FORCE:
+                LOG_ERROR("Forcing VHDL variables is not suported by the FLI");
```
Comment (cmarqu) [inline cocotb/share/lib/fli/FliObjHdl.cpp:258]:
```suggestion
                LOG_ERROR("Forcing VHDL variables is not supported by the FLI");
```
(and in the rest of the file :) ).

**Comment by cmarqu** [comment]
This feature is worth a newsfragment.

File: documentation/source/newsfragments/2775.feature.rst [hunk 5]
```diff
@@ -0,0 +1 @@
+:class:`~cocotb.handle.Force`, :class:`~cocotb.handle.Freeze`, and :class:`~cocotb.handle.Release` are now supported when using the FLI.
```
Comment (cmarqu) [inline documentation/source/newsfragments/2775.feature.rst:1]:
```suggestion
:class:`~cocotb.handle.Force`, :class:`~cocotb.handle.Freeze`, and :class:`~cocotb.handle.Release` are now supported when using the FLI, Questa's traditional method to access VHDL.
```

**Review by cmarqu — APPROVED** [review]
